components/authz-service/storage/v2/postgres/postgres.go

```diff
@@ -1338,12 +1338,29 @@ func (p *pg) Pristine(ctx context.Context) error {
 	return p.recordMigrationStatus(ctx, enumPristine)
 }
 
+func (p *pg) recordMigrationStatusAndNotifyPG(ctx context.Context, ms string) error {
+	tx, err := p.db.BeginTx(ctx, nil /* use driver default */)
```

Comment:
Probably a naive question: don't you need a "defer _some-kind-of-rollback_" on this transaction? I don't see it anywhere else this construct is used, so I'm guessing the answer is no, but why is that?

Reply:
It's all in the `ctx`, see this note: https://github.com/chef/automate/blob/02fcff1a6660ba1d3f1fe847a5d9785c744afa79/components/authz-service/storage/v2/postgres/postgres.go#L58-L63

Now, whether not cancelling this myself is a problem or not (I haven't put a `ctx, cancel := context.WithCancel(ctx); defer cancel()` in this function), I'll figure out now.

Reply:
Update: We don't need to cancel ourselves, grpc takes care of this -- see this toy experiment https://gist.github.com/srenatus/ce12b31ea517f16c024e4f8736fa5f2b

Reply:
https://github.com/chef/automate/pull/501/commits/329e8ef6fbef7337c3d5161ebd3b9a00cca23dbd Added a cleanup accordingly.

Reply:
And reverted the cleanup. Experiment must have been flawed, we _need_ the `context.WithCancel`.